internal/jujuclient/dial.go

```diff
@@ -230,7 +273,17 @@ func (c *Connection) redial(ctx context.Context, requiredPermissions map[string]
 	if dialCount > 10 {
 		return errors.E(op, "dial count exceeded")
 	}
-	api, err := c.dialer.Dial(ctx, c.ctl, c.mt, requiredPermissions)
+
+	grantedPermission := make(map[string]string)
+	for resource, permission := range requiredPermissions {
+		err := c.user.CheckPermission(ctx, resource, permission)
+		if err != nil {
+			return err
+		}
+		grantedPermission[resource] = permission
+	}
```

Comment:
Can we add a comment above on why we check the permission. We discussed it in a previous PR and I've forgotten and it still doesn't strike me as obvious.

Reply:
done